### PR TITLE
added transformation services as features

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/core/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java [hunk 1]
```diff
@@ -48,6 +49,7 @@ protected void unsetFeaturesService(FeaturesService featureService) {
 
     @Override
     public List<Extension> getExtensions(Locale locale) {
+
```
Comment (maggu2810) [inline bundles/core/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java]: Would you like to start the function with an empty line?

**Comment by maggu2810** [comment]
I would suggest to create a pull request for shk for ESH stuff and wrap that features in OH (as it is done e.g. for shk-esh-base, sonos binding, etc.).
So we could stop do the same work multiple times.
I have done the automation integration just this evening, so I added the tranformation ones, too.

**Comment by kaikreuzer** [comment]
Ok, better now?

> I would suggest to create a pull request for shk for ESH stuff

I would actually suggest to create a pull request from shk to ESH for that stuff - then the features would be directly build and made available with an ESH build. Is this the plan as soon as we can do versioned releases?

**Comment by maggu2810** [comment]
If shk is hosted at Eclipse and we declare a bundle dependency in a feature, is there a CQ necessary for everyone?
It is not for the framework, it is for a product that uses the framework.
All dependencies e.g. Pax Web and all the other Karaf feature we depend on have to be checked the same way as we have to check if something can be used as a work-with dependency (e.g. as you have done with nrjavaserial IIRC).

**Comment by kaikreuzer** [comment]
If we only build the features and don't do the validation (which could be done in the products), then the build would not have a dependency on e.g. pax-web at all, right? As the features are no code by themselves, they also don't have any inherent dependency - they can be considered as a kind of documentation. I would claim that we do not even need to declare build time dependencies for anything here.

**Comment by kaikreuzer** [comment]
Rebased after merging #560.

**Pull request merged**